core/libs/test_utils/src/lib.rs

```diff
@@ -270,37 +271,31 @@ pub fn assert_repo_integrity(core: &Core) {
 }
 
 pub fn doc_repo_get_all(config: &Config, source: RepoSource) -> Vec<EncryptedDocument> {
-    dump_local_storage::<_, Vec<u8>>(config, document_repo::namespace(source))
-        .into_iter()
-        .map(|s| bincode::deserialize(s.as_ref()).unwrap())
-        .collect::<Vec<EncryptedDocument>>()
-        .into_iter()
-        .collect()
+    let mut docs = vec![];
+    for file in list_files(config, &source) {
+        let content = fs::read(file).unwrap();
+        docs.push(bincode::deserialize(&content).unwrap());
+    }
+    docs
 }
 
-fn dump_local_storage<N, V>(db: &Config, namespace: N) -> Vec<V>
-where
-    N: AsRef<[u8]> + Copy,
-    V: From<Vec<u8>>,
-{
-    let path_str = local_storage::namespace_path(db, namespace);
-    let path = Path::new(&path_str);
+fn list_files(db: &Config, namespace: &RepoSource) -> Vec<String> {
+    let path = document_repo::namespace_path(db, namespace);
+    let path = Path::new(&path);
 
-    match fs::read_dir(path) {
+    match fs::read_dir(&path) {
         Ok(rd) => {
             let mut file_names = rd
                 .map(|dir_entry| dir_entry.unwrap().file_name().into_string().unwrap())
+                .map(|entry| {
```

Comment:
No need for two separate, consecutive map calls, right?

Reply:
oh I thought it's conceptually nicer

Reply:
Imo a single map function is nicer

Reply:
[d4867e9](https://github.com/lockbook/lockbook/pull/1259/commits/d4867e99e6c55765e0c4eceab4dbe9d1bfc68c75)